Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs

```diff
@@ -110,231 +123,227 @@ private void OnPlayerSpawningEvent(PlayerSpawnCompleteEvent ev)
-        // }
-
-        if (_map.TryLoad(mapId, arena, out var depotUid5s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(2200f, 4200f)
-            }))
-        {
-            var meta = EnsureComp<MetaDataComponent>(depotUid5s[0]);
-            _meta.SetEntityName(depotUid5s[0], "The Pit", meta);
-            _shuttle.SetIFFColor(depotUid5s[0], civilianColor);
-        }
+        var mapId = GameTicker.DefaultMap;
+        var dungenTypes = _prototypeManager.EnumeratePrototypes<DungeonConfigPrototype>();
 
-        if (_map.TryLoad(mapId, cove, out var depotUid6s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(10000f, 15000f)
-            }))
+        foreach (var dunGen in dungenTypes)
         {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("Cove", out var stationProto))
+
+            var seed = _random.Next();
+            var offset = _random.NextVector2(3000f, 8500f);
+            if (!_map.TryLoad(mapId, "/Maps/_NF/Dungeon/spaceplatform.yml", out var grids, new MapLoadOptions
+                {
+                    Offset = offset
+                }))
             {
-                _station.InitializeNewStation(stationProto.Stations["Cove"], depotUid6s);
+                continue;
             }
 
-            var meta = EnsureComp<MetaDataComponent>(depotUid6s[0]);
-            _meta.SetEntityName(depotUid6s[0], "Pirate's Cove", meta);
-            _shuttle.SetIFFColor(depotUid6s[0], lpbravoColor);
-            _shuttle.AddIFFFlag(depotUid6s[0], IFFFlags.HideLabel);
+            var mapGrid = EnsureComp<MapGridComponent>(grids[0]);
+            _shuttle.AddIFFFlag(grids[0], IFFFlags.HideLabel);
+            _console.WriteLine(null, $"dungeon spawned at {offset}");
+            offset = new Vector2i(0, 0);
+
+            //pls fit the grid I beg, this is so hacky
+            //its better now but i think i need to do a normalization pass on the dungeon configs
+            //because they are all offset. confirmed good size grid, just need to fix all the offsets.
+            _dunGen.GenerateDungeon(dunGen, grids[0], mapGrid, (Vector2i) offset, seed);
         }
+    }
 
-        if (_map.TryLoad(mapId, lodge, out var lodgeUids, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(1650f, 3400f)
-            }))
+    private void GenerateDepots(IEnumerable<PointOfInterestPrototype> depotPrototypes)
+    {
+        //For depots, we want them to fill a circular type dystance formula to try to keep them as far apart as possible
+        //Therefore, we will be taking our range properties and treating them as magnitudes of a direction vector divided
+        //by the number of depots set in our corresponding cvar
+        var protoList = depotPrototypes.ToList();
+        var depotCount = _configurationManager.GetCVar(NF14CVars.CargoDepots);
+        var rotation = Math.PI / depotCount;
```

Comment:
If you want to cover a whole circle, rather than half of it, use 2pi.
```suggestion
        var rotation = 2 * Math.PI / depotCount;
```